### DISCUSSION TITLE
how are buckets resolved

### DISCUSSION THREAD

**Discussion by Yakiyo** [ScoopInstaller/Scoop#5624] [Help and Questions] [answered]
I wanted to ask how scoop buckets are structured. Almost all of the buckets ive seen follow the structure of the bucket template. In them, all manifests are stored inside the `bucket/` directory in the root of the repo. but is that the only way buckets must be made? does scoop resolve manifests in nested directories or maybe if the manifests are in the root of the repo. Would those work too?

**Comment by HUMORCE** [comment, marked as answer]
> does scoop resolve manifests in nested directories

Yes, https://github.com/ScoopInstaller/Scoop/pull/5119

> maybe if the manifests are in the root of the repo

No, `<bucket>\bucket` hard code in https://github.com/ScoopInstaller/Scoop/blob/master/lib/buckets.ps1.

**Reply by Yakiyo** [reply]
i see, so scoop recursively searches in `{{bucket}}/bucket` dir. thank you for answering

**Reply by markembling** [reply]
> > maybe if the manifests are in the root of the repo
> 
> No, `<bucket>\bucket` hard code in https://github.com/ScoopInstaller/Scoop/blob/master/lib/buckets.ps1.

Not quite. It has this code (line 24 onwards):

```powershell
$bucket = "$bucketsdir\$Name"
if ((Test-Path "$bucket\bucket") -and !$Root) {
    $bucket = "$bucket\bucket"
}
```

So if there is a `bucket` directory, it is used (and all manifests are expected to be found within it). If there isn't, it uses the root of the bucket repo. It does look like the `bucket/` directory is the recommended convention though.
